monkey/common/network/network_range.py

```diff
@@ -77,7 +80,21 @@ def validate_range(address_str: str):
             raise InvalidNetworkRangeError(e)
 
     @staticmethod
-    def check_if_range(address_str):
+    def check_if_hostname(hostname: str):
+        if len(hostname) > 253 or hostname[-1] == ".":
+            return False
+
+        labels = hostname.split(".")
+
+        # the TLD must be not all-numeric
+        if re.match(r"[0-9]+$", labels[-1]):
```

Comment:
This regex tests that the TLD cannot end with digits.

```suggestion
        if re.match(r"^[0-9]+$", labels[-1]):
```
There's a test that I think should fail given the current code. I'm not sure why it doesn't.

Also consider compiling this regex only once as a class variable.